src/coreclr/src/jit/lclvars.cpp

```diff
@@ -5084,12 +5123,18 @@ void Compiler::lvaFixVirtualFrameOffsets()
         // Is this a non-param promoted struct field?
         //   if so then set doAssignStkOffs to false.
         //
-        if (varDsc->lvIsStructField && !varDsc->lvIsParam)
+        if (varDsc->lvIsStructField)
         {
             LclVarDsc*       parentvarDsc  = &lvaTable[varDsc->lvParentLcl];
             lvaPromotionType promotionType = lvaGetPromotionType(parentvarDsc);
 
-            if (promotionType == PROMOTION_TYPE_DEPENDENT)
+#if defined(TARGET_X86)
+            // On x86, we set the stack offset for a promoted field
+            // to match a struct parameter in lvAssignFrameOffsetsToPromotedStructs.
+            if ((!varDsc->lvIsParam || parentvarDsc->lvIsParam) && promotionType == PROMOTION_TYPE_DEPENDENT)
```

Comment:
I was seeing cases where this wasn't true while crossgenning System.Private.CoreLib. I'll re-investigate to see if I still need to do this.

Reply:
Not including this case didn't cause any issues, but it caused the bogus offset to be adjusted and then dumped in the JIT dump. Since the bogus offset gets overridden in the next step of stack offset assignment to match the struct parameter, it isn't really a problem, but it's confusing in the jit dump.